mesonbuild/build.py

```diff
@@ -450,7 +450,7 @@ def __init__(self, name, subdir, subproject, is_cross, sources, objects, environ
         self.is_unity = unity_opt == 'on' or (unity_opt == 'subprojects' and subproject != '')
         self.environment = environment
         self.sources = []
-        self.compilers = OrderedDict()
+        self.compilers = OrderedDict() # type: OrderedDict[str, Compiler]
```

Comment:
this needs to be typing.OrderedDict or mypy will complain.

Reply:
Actually it didn't:
```
mesonbuild/build.py:452: error: Need type annotation for 'sources'
mesonbuild/build.py:454: error: Need type annotation for 'objects'
```

Reply:
`mypy --strict --follow-imports silent --python-version 3.5 mesonbuild/build.py`. Tried again with `--python-version 3.5` this time, and 3.5-built mypy, and still no problem with line 453 just like above.